### PR TITLE
Change invited admin user password on first login

### PR DESCRIPTION
**Pull request by ji-mmyliu**
## Notion ticket link

[Change Password on First Login for Invited User](https://www.notion.so/uwblueprintexecs/Change-Password-on-First-Login-for-Invited-User-7eab8e395ae44fb9a1c3c109a05443ad)

## Implementation description
* Used Firebase's REST API to change password using user's access token.
* Split logic for change password and updating user status into two separate endpoints.

## Steps to test
1. Post the invite admin endpoint in the backend using Postman and a manually obtained admin access token to invite a new admin user
2. Verify and login using the new credentials
3. Choose a new password and check that your user is now `Active`.

## What should reviewers focus on?
* The process flow and potential security issues.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by carolynzhang18** [comment]
WORKS SUPER NICE I LOVE IT!!! 🤩

**Comment by carolynzhang18** [comment]
One thing I just thought of - let's say I'm logging in for the first time, I reach the "change your password" page, and then before changing my password I close the tab (so I didn't actually fill in the form).
Next time I open the app, I'm still on the "change your password" page. Do you think that's an issue? I thiiink it's ok but we can discuss at first meeting :)

File: backend/services/implementations/authService.ts [hunk 1]
```diff
@@ -16,12 +18,19 @@ class AuthService implements IAuthService {
...
     userService: IUserService,
     emailService: IEmailService | null = null,
   ) {
     this.userService = userService;
     this.emailService = emailService;
+    const firebaseApp = firebase.initializeApp({
+      apiKey: process.env.FIREBASE_WEB_API_KEY,
+      projectId: process.env.FIREBASE_PROJECT_ID,
+    });
+    this.firebaseAuth = firebaseAuth.getAuth(firebaseApp);
```
Comment (carolynzhang18) [inline backend/services/implementations/authService.ts:33]: Are any of these Firebase things still being used in `AuthService`?

Comment (carolynzhang18) [reply backend/services/implementations/authService.ts:33]: If not, could we remove them from here + uninstall them (this should update `backend/package.json` and `backend/yarn.lock` as well)?

File: backend/package.json [hunk 2]
```diff
@@ -25,6 +27,7 @@
     "dotenv": "^8.2.0",
     "express": "^4.19.2",
     "express-rate-limit": "^6.2.0",
+    "firebase": "^10.12.3",
     "firebase-admin": "^9.5.0",
```
Comment (carolynzhang18) [inline backend/package.json:31]: (see comment below)
If the 3 firebase packages you installed aren't actively being used anymore, could we uninstall them?

**Review by carolynzhang18 — COMMENTED** [review]
awesome awesome stuff, thanks for your hard work with this one!!! 😎
left a few minor comments but everything else looks good to me!

File: docker-compose.yml [hunk 3]
```diff
@@ -29,6 +32,18 @@ services:
...
+    image: mongo
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=eaf_user
+      - MONGO_INITDB_ROOT_PASSWORD=eaf_password
+    ports:
+      - 27017:27017
+    volumes:
+      - data:/data/db
+    healthcheck:
+      test: echo 'db.runCommand("ping").ok' | mongosh localhost:27017/test --quiet
```
[the remark this answers is not in the document]
Comment (ji-mmyliu) [reply docker-compose.yml:46]: OOPS! Yep didn't mean to push that 🙃

**Review by carolynzhang18 — APPROVED** [review]

**Review by mxc-maggiechen — APPROVED** [review]
Works great!

**Pull request merged**